library/src/main/java/io/ona/kujaku/services/TrackingService.java

```diff
@@ -817,18 +815,33 @@ private int getNumberOfLocationsRecorded() {
      *
      * @return
      */
-    public List<Location> getRecordedLocations() {
-        return this.recordedLocations;
+    public List<KujakuLocation> getRecordedKujakuLocations() {
+        return this.recordedKujakuLocations;
+    }
+
+    /**
+     * Set Tag for next points
+     *
+     * @param tag
+     */
+    public void setTag(long tag) {
+        this.trackingServiceOptions.setTag(tag);
     }
 
     /**
-     * Record pending Location
+     *  Record pending Location
+     *
+     * @param tag
      */
-    public void takeLocation () {
-        if (pendingRecordingLocation != null) {
-            Location pendingLocation = pendingRecordingLocation;
+    public void takeLocation (long tag) {
+        if (pendingRecordingKujakuLocation != null) {
+            KujakuLocation pendingLocation = new KujakuLocation(pendingRecordingKujakuLocation, pendingRecordingKujakuLocation.getTag());
```

Comment:
Hi Ephraim, the only part of the code where it nullify the `pendingRecordingKujakuLocation` is an initialization or when we record the pendingRecordingKujakuLocation. 
The tag used in the `takeLocation (long tag)` function is not sticky at all. It is only used / forced when the app call this function. Otherwise, the tag used is the initial tag defined in the `TrackingServiceOptions` instance

Reply:
When the `pendingRecordingKujakuLocation` is null, it means that the GPS has'nt provided a new Location yet.

Reply:
Cool, so `takeLocation` worked without the tag being sticky. However, I clicked on the button 6 times and the tag was only added to 2 locations. I think this is a bit flaky and does not serve the purpose. My test case scenario is that this would be clicked when stationary i.e. when the user confirms that the location can be tagged(The location meets the criteria). Is there a way to solve this?

Reply:
Thanks Ephraim, i will check how to solve this

Reply:
Ok the solution is to use the last best location variable if the pendinglocation is null. If the user wants to force taking a location, if the GPS did'nt send a new location, the system will use the last best location. This one will never be null.

Reply:
Cool, I will test this out

Reply:
It now works OK

Reply:
Awesome